### PR TITLE
WB-1173.4: Add states to BasicCell (hovered, focused, pressed, active, disabled)

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:
- Adds the cell states (hovered, focused, pressed, active, disabled).
- Adds `onClick` to include a pressable cell.
- Adds the `aria-label` prop.

### Plan:
1. Add wonder-blocks-cell package #1273
2. Add BasicCell component layout (`title, leftAccessory, leftAccessoryStyle, rightAccessory, rightAccessoryStyle`). #1275
3. Add `horizontalRule, style, testId`.  1276
4. **Add states to BasicCell (hovered, focused, pressed, active, disabled) [CURRENT]**

Issue: WB-1173

## Test plan:

- Verify the visual snapshots in Chromatic.
- Also verify that the Storybook instance renders all the stories and
  includes the documentation properly.

[BasicCell active](https://5e1bf4b385e3fb0020b7073c-fwzsprnuan.chromatic.com/?path=/story/cell-basiccell--basic-cell-active)
Screen Shot 2021-12-23 at 5 08 16 PM

[BasicCell disabled](https://5e1bf4b385e3fb0020b7073c-fwzsprnuan.chromatic.com/?path=/story/cell-basiccell--basic-cell-disabled)
Screen Shot 2021-12-23 at 5 09 19 PM

### CODE REVIEW

File: packages/wonder-blocks-cell/src/components/internal/cell-core.js [hunk 1]
```diff
@@ -127,6 +184,50 @@ const styles = StyleSheet.create({
         // content is too short.
         marginLeft: "auto",
     },
+
+    /**
+     * States
+     */
+    hovered: {
```
Comment (jeresig) [inline packages/wonder-blocks-cell/src/components/internal/cell-core.js:191]: I think the test coverage dropped as there's no tests for some of the states here?

**Review by jeresig — APPROVED** [review]

**Review by lillialexis — APPROVED** [review]
When I'm in the Story, and I try and change stuff, nothing happens. Am I doing something wrong? Or is it broken?

**Comment by jandrade** [comment]
> When I'm in the Story, and I try and change stuff, nothing happens. Am I doing something wrong? Or is it broken?

@lillialexis  I only configured the first story to be interactive, but probably can do the same with other stories where it makes sense :).

**Comment by jandrade** [comment]
The parent pull-request ([#1276](https://github.com/Khan/wonder-blocks/pull/1276)) has been merged into `feature/cell`, but this branch (`cell-b4`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

**Pull request merged**